api/v1beta1/azurecluster_webhook.go

```diff
@@ -63,13 +63,6 @@ func (c *AzureCluster) ValidateUpdate(oldRaw runtime.Object) (admission.Warnings
 		allErrs = append(allErrs, err)
 	}
 
-	if err := webhookutils.ValidateImmutable(
```

Comment:
Rather than allow mutability, should we simply allow changes from `""` to `"<non-empty string>"`

Reply:
One other option might be to bring back the global config only for sub ID.

Reply:
Good question.

That allows the field to be updated at most one time.

What happens if the value is wrong, e.g. by mistake?

Can the user fix the mistake? For example, can they pause the Cluster, remove the AzureCluster resource, fix the subscriptionID, and re-create it?

Reply:
> One other option might be to bring back the global config only for sub ID.

I think we're discussing whether making this field mutable introduces risk to users.

Let's please remember that the global config is (and always has been) mutable.

So, making the field write-many does not introduce more risk for users.
 
Moreover, making the field write-once allows users to migrate away from the write-many global config, actually reducing risk.

Reply:
What other information can I provide to help us decide between write-once and write-many?

Reply:
I think the goals are:
1. Allow users who previously relied on the manager credential bootstrap for the  subscription ID to migrate to CAPZ that no longer uses this credential.
2. Prevent users from inadvertently changing the subscription ID.

Reply:
I don't see a solution that meets both (1) and (2) perfectly.

* Write-once is a problem for (1) if the user makes a mistake, but helps with (2).
* Write-many works for (1), but does not help with (2).

How about something unusual, like "conditional" mutability. For example, if the user wants to change subscriptionID, they have to first set an annotation on AzureCluster. When the annotation is present, subscriptionID is mutable. After the user sets subscriptionID, they remove the annotation. At that point, subscriptionID is again immutable.

Reply:
The above idea is in 46bfbafd7543f8ce860aaf30315dbd3a9af7c3c4.

Reply:
Once we have consensus, I'll of course update the docs in this PR.